internal/app/machined/pkg/controllers/network/rtestutils/assert.go

```diff
@@ -0,0 +1,160 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+// Package rtestutils provides utilities for testing controllers. Fork of rtestutils from cosi-project/runtime.
+package rtestutils
+
+import (
+	"context"
+	"fmt"
+	"sort"
+	"strings"
+	"testing"
+
+	"github.com/cosi-project/runtime/pkg/resource"
+	"github.com/cosi-project/runtime/pkg/resource/rtestutils"
+	"github.com/cosi-project/runtime/pkg/safe"
+	"github.com/cosi-project/runtime/pkg/state"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+// ResourceWithRD is a resource providing resource definition.
+type ResourceWithRD = rtestutils.ResourceWithRD
+
+// AssertNoResource asserts that a resource no longer exists.
+func AssertNoResource[R ResourceWithRD](ctx context.Context, t *testing.T, st state.State, id resource.ID) {
+	rtestutils.AssertNoResource[R](ctx, t, st, id)
+}
+
+// AssertResources asserts on a resource list.
+func AssertResources[R ResourceWithRD](
+	ctx context.Context,
+	t *testing.T,
+	st state.State,
+	ids []resource.ID,
+	assertionFunc func(r R, assertion *assert.Assertions),
+	opts ...Option,
```

Comment:
I'm not sure if we have any other places that require us to pass Namespace separately, and those I don't know if it make sense to move it into runtime project.

Reply:
I think certainly we shouldn't have a copy here

Reply:
Okay - moved to https://github.com/cosi-project/runtime/pull/236